CMakeLists.txt

```diff
@@ -40,19 +28,32 @@ else()
   message("CGAL not found, can not use new quad detector")
 endif()
 
-set(CMAKE_C_FLAGS "-Wall -g")
-set(CMAKE_CXX_FLAGS "-Wall -g")
-
-if (APPLE OR UNIX)
-set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wsign-compare -frounding-math")
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wsign-compare -frounding-math")
-endif()
-
-set(CMAKE_C_FLAGS_DEBUG "-O")
-set(CMAKE_CXX_FLAGS_DEBUG "-O")
-
-set(CMAKE_C_FLAGS_RELEASE "-O2")
-set(CMAKE_CXX_FLAGS_RELEASE "-O2")
 
-add_subdirectory(src)
+find_package(OpenCV REQUIRED)
+include_directories(
+  include/${PROJECT_NAME}
+  ${OpenCV_INCLUDE_DIRS}
+  ${catkin_INCLUDE_DIRS}
+)
+
+FILE(GLOB_RECURSE apriltags_swatbotics_headers "include/*.h")
+
+add_library(${PROJECT_NAME} 
+  src/CameraUtil.cpp
+  src/DebugImage.cpp
+  src/Geometry.cpp 
+  src/GrayModel.cpp
+  src/MathUtil.cpp 
+  src/Refine.cpp
+  src/TagDetector.cpp 
+  src/TagFamily.cpp
+  src/TagFamilies.cpp
+  src/UnionFindSimple.cpp
+  ${apriltags_swatbotics_headers}
```

Comment:
I don't think we actually need to include the headers in the library declaration.